nexus_constructor/nexus/nexus_wrapper.py

```diff
@@ -141,6 +143,20 @@ def create_nx_group(
         self._emit_file()
         return group
 
+    @staticmethod
+    def duplicate_nx_group(
+        group_to_duplicate: h5py.Group, components_list
+    ) -> h5py.Group:
+        new_group_name = generate_unique_name(
```

Comment:
I think this should be done in the duplicate method in component and the destination name passed as a input arg to this method. `duplicate_nx_group` may then be useful in other situations.